Doc improvs (#17)
nrfconnect/nrf70-bm: nrf70_bm_lib/docs/source/nrf70_bm_lib.rst

  .. note::
     In the reference implementation for Zephyr tasklet work is offloaded to Zephyr kernel workqueues.

> Optimzing scan operation

Reviewer: Fix typo

nRF70 Series device states
##########################

The power save state of the nRF70 Series device is described through a combination of the physical power state of the logic or circuits and the logical functional state as observed by 802.11 protocol operations.

Power state
***********

The nRF70 Series device can be in one of the following power states:

* **Active:** The device is **ON** constantly so that it can receive and transmit the data.
* **Sleep:** The device is **OFF** to the majority of the blocks that cannot receive and transmit the data.
  In this state, the device consumes low power (~15 µA).
  Real-time Clock (RTC) domain, RF retention memory, and firmware retention memory are powered **ON** to retain the state information.
* **Shutdown:** The device is completely powered **OFF**.
  In this state, the device consumes very low power (~2 µA) and does not retain any state information (apart from the values in the OTP memory).
  The device will only respond to a BUCKEN assertion to wake from the Shutdown state.

.. note::

>   The **Sleep** state is only applicable when ``CONFIG_NRF_WIFI_LOW_POWER`` is enabled.

Reviewer: Should we give a more "friendly" description, i.e. from the user's perspective?

Author: `The **Sleep** state is only applicable when the Zephyr application is built with the Kconfig option ``CONFIG_NRF_WIFI_LOW_POWER`` enabled.`

How about this?

>   The **Sleep** state is only applicable when ``CONFIG_NRF_WIFI_LOW_POWER`` is enabled.

Reviewer: ```suggestion
  To allow the nRF70 Series device enter the **Sleep** state when applicable, the ``CONFIG_NRF_WIFI_LOW_POWER`` Kconfig option must be enabled.
```

> The nRF70 Series transition to and from the Shutdown state is automatically managed by the nRF Wi-Fi driver.

Reviewer: ```suggestion
The nRF70 Series transition to and from the **Shutdown** state is automatically managed by the nRF Wi-Fi driver.
```

> When the network interface is brought down, the nRF Wi-Fi driver puts the nRF70 Series device in Shutdown state.

Reviewer: @krish2718  is this something that is OS-agnostic, or is it something related to the nRF Wi-Fi driver and NET Management (Zephyr) only? If it is Zephyr-specific we need to get this explained.

Author: It's kind of both, the concept of `interface` is Zephyr, but OSAL uses `device` and during `interface down`, `device` is removed which turns off the power.

> When the network interface is brought up, the nRF Wi-Fi driver puts the nRF70 Series device in Active state.

Reviewer: Same comment here. Instead of network interface, should we refer to BM initialization functions?

Functional state
****************

> The functional state information of the nRF70 Series device is described through the following states:

Reviewer: ```suggestion
In terms of functionality, the nRF70 Series device can reside in the following states:
```

> * **Scanning:** The device is in the scanning state, it is **Active** and is scanning for the available networks.

Author: Scan doesn't mean RX only, we also have Active scan which is TX.

> * **Scanning:** The device is in the scanning state, it is **Active** and is scanning for the available networks.

Reviewer: ok
